handler/src/main/java/io/netty/handler/ssl/DefaultOpenSslSession.java

```diff
@@ -49,7 +51,7 @@
     private volatile int packetBufferSize = ReferenceCountedOpenSslEngine.MAX_RECORD_SIZE;
     private volatile long lastAccessed;
     private volatile Certificate[] localCertificateChain;
-    private volatile boolean invalid;
+    private final AtomicBoolean invalid = new AtomicBoolean(false);
```

Comment:
Couldn't we keep the volatile field and use AtomicIntegerFieldUpdater?

Reply:
done